data-prepper-plugins/mutate-event-processors/src/main/java/org/opensearch/dataprepper/plugins/processor/mutateevent/ConvertEntryTypeProcessor.java

```diff
@@ -80,7 +83,18 @@ public Collection<Record<Event>> doExecute(final Collection<Record<Event>> recor
                     if (keyVal != null) {
                         if (!nullValues.contains(keyVal.toString())) {
                             try {
-                                recordEvent.put(key, converter.convert(keyVal, converterArguments));
+                                if (keyVal instanceof ArrayList || keyVal.getClass().isArray()) {
+                                    ArrayList<Object> inputList;
```

Comment:
Rather than use a `List` or an `ArrayList`, you can use a `Stream`. Then you can avoid the list conversions below.

```
if (keyVal.getClass().isArray()) {
  inputStream = Arrays.stream((Object[])keyVal);
} else {
  inputStream = ((List<?>)keyVal).stream();
}
List<?> replacementList = inputStream.map(i -> converter.convert(i, converterArguments)).collect(Collectors.toList());
recordEvent.put(key, replacementList);
```

Also, you do a lot on line 93. I think splitting this up would be cleaner, so I showed that in my example above.

Reply:
More specifically, you can avoid creating a brand new `List` that you don't need. Using the `Stream` will not copy the list.